lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -50,43 +55,136 @@ define((require, exports, module) ->
       )
       return
 
-    callbackHidePopover = ->
-      popoverHandler.hide($("#formula"))
-      editor.off("changeSelection", callbackHidePopover)
-      editor.session.off("changeScrollTop", callbackHidePopover)
-      editor.session.off("changeScrollLeft", callbackHidePopover)
-      return
+    jqEditorContainer = $(editor.container)
+    jqFormula = -> $("#formula")
+
+    ch = contextHandler = {
+      contextPreviewExists: false
+      updateDelay: 1000
+
+      removeRegex: /\\end\{equation\}|\\begin\{equation\}|\\label\{[^\}]*\}/g
 
-    onLoaded = ->
-      try
-        content = katex.renderToString(
-          editor.getSelectedText(),
+      getEquationRange: (cursorRow) ->
+        i = cursorRow
+        while latexContextParser.getContext(editor.session, i - 1) == "equation"
+          i -= 1
+        start = i
```

Comment:
assign `i = cursorRow` again, to skip checking the rows which have already been checked?